src/bootstrap/replication_client_test.rs

```diff
@@ -0,0 +1,206 @@
+#[cfg(test)]
+mod tests {
+    use crate::{
+        bootstrap::service::{ReplicatorBootstrap, WorkUnitResponse},
+        cfg::Config,
+        proto::{self, replication_service_server::ReplicationServiceServer},
+        replication::{
+            replication_test_utils::replication_test_utils::*,
+            replicator::{self},
+        },
+        storage::{
+            db::RocksDB,
+            store::{
+                engine::PostCommitMessage,
+                test_helper::{self},
+            },
+            trie::merkle_trie::MerkleTrie,
+        },
+        utils::factory::{self, messages_factory},
+    };
+    use std::time::Duration;
+    use tempfile::TempDir;
+    use tonic::transport::Server;
+
+    #[tokio::test]
+    async fn test_replication_client_bootstrap() {
```

Comment:
Is there any scenario where we may reprocess messages and trie keys after they've already been committed? If so, can we add a test to make sure the deduplication logic works properly?

Reply:
Yes, it does handle duplicates properly, but I will add a test of this. Actually, there are other scenarios we should test, like an interrupted connection etc... I'll add these tests in the next PR, since I have some test utility classes there. Your comment about the testing is a good one - We should have more extensive tests, since there are many edge cases that get handled. I'll add all these into the next PR